libyul/backends/evm/ControlFlowGraph.h

```diff
@@ -210,10 +212,12 @@ struct CFG
 	{
 		std::shared_ptr<DebugData const> debugData;
 		Scope::Function const& function;
+		FunctionDefinition const& functionDefinition;
 		BasicBlock* entry = nullptr;
 		std::vector<VariableSlot> parameters;
 		std::vector<VariableSlot> returnVariables;
 		std::vector<BasicBlock*> exits;
+		bool canContinue = true;
```

Comment:
Yes, ``FunctionInfo`` refers to a function definition - but calling it ``FunctionDefinition`` as well here would probably be even more confusing, wouldn't it? On the other hand, the same could be said for ``FunctionCall``, so maybe it'd indeed have been better to call it ``FunctionDefinition``...

I see two fields in common: ``function`` and ``canContinue`` - not sure there's that much need to unify them :-).
The original reason against just having ``FunctionCall`` just refer to ``FunctionInfo`` is that the original versions of the ``ControlFlowGraphBuilder`` did a straight-forward direct traversal of the Yul AST, in which function calls can refer to functions that are only defined later in the block (so I wouldn't have had the ``FunctionInfo`` available when constructing the ``FunctionCall``). Since then, however, we actually collect all functions defined in a block up front, so actually now we could indeed change this...

Reply:
I'd tend towards not changing this in this same PR, though, in order not to complicate it further.

Reply:
Yeah, I wouldn't touch it in the same PR.

But might still be worth changing eventually. Now you have two `canContinue` fields and this one has a default so you could easily forget to set it explicitly to the right value.

Reply:
And as far as naming goes. Maybe something like `FunctionDefinitionMetadata` or `FunctionDefinitionInfo` would have been clearer? But again, probably not something to change in this PR.